src/array.rs

```diff
@@ -110,7 +111,18 @@ impl<'a, const P: usize, const W: usize> RepresentationTrait for Array<'a, P, W>
     /// Return memory size of `Array` representation
     #[inline]
     fn size_of(&self) -> usize {
-        size_of::<usize>() + size_of_val(self.arr)
+        // The size of the len attribute, which is an usize
+        let len_size = size_of::<usize>();
+        // The size of the cap attribute, which is an usize
+        let cap_size = size_of::<usize>();
+        // The size of the arr attribute, which is a slice
+        // and therefore composed of a pointer (usize) and a length (usize)
+        let arr_size = size_of::<usize>() * 2;
```

Comment:
Not sure I understand the "multiply by 2" aspect of this logic, however I don't think we should measure memory usage of `Array` and other structs except `CardinalityEstimator` since these are stack allocated and not actually stored / exposed beyond `pub(crate)` visibility.